### ISSUE TITLE
can't reproduce the result of BoT

### ISSUE THREAD

**Issue by Theohhhu** [JDAI-CV/fast-reid#47]
Hi Liao,
Thanks for your awesome implementation.
I met some problem when trying to reproduce the result showed in your model_zoo
e.g. 
As for market1501, strictly follow your step without any modification using your script, I only get 92.5 R@1 using resnet and  93.3 R@1 using resnet-ibn-a. Seems like there are some problems in hyperparameters. Could you check it(resnet+market1501+Bot) for me please?

Thanks in advance. 
Here I only change the max iteration to 28000 as I found there is small improvement on final performance after 18000 iterations.

MODEL:
  HEADS:
    NUM_CLASSES: 751

SOLVER:
  MAX_ITER: 28000

  STEPS: [8000, 14000]

  WARMUP_ITERS: 2000

DATASETS:
  NAMES: ("Market1501",)
  TESTS: ("Market1501",)

OUTPUT_DIR: "logs/market1501/bagtricks"

[32m[04/30 01:33:40 fastreid]: [0mRunning with full config:
CUDNN_BENCHMARK: True
DATALOADER:
  NUM_INSTANCE: 4
  NUM_WORKERS: 16
  PK_SAMPLER: True
DATASETS:
  NAMES: ('Market1501',)
  TESTS: ('Market1501',)
INPUT:
  DO_AUGMIX: False
  DO_AUTOAUG: False
  DO_CJ: False
  DO_FLIP: True
  DO_PAD: True
  FLIP_PROB: 0.5
  PADDING: 10
  PADDING_MODE: constant
  REA:
    ENABLED: True
    MEAN: [123.675, 116.28, 103.53]
    PROB: 0.5
  RPT:
    ENABLED: False
    PROB: 0.5
  SIZE_TEST: [256, 128]
  SIZE_TRAIN: [256, 128]
MODEL:
  BACKBONE:
    DEPTH: 50
    LAST_STRIDE: 1
    NAME: build_resnet_backbone
    PRETRAIN: True
    WITH_IBN: False
    WITH_NL: False
    WITH_SE: False
  HEADS:
    CLS_LAYER: linear
    IN_FEAT: 2048
    MARGIN: 0.15
    NAME: BNneckHead
    NUM_CLASSES: 751
    POOL_LAYER: avgpool
    REDUCTION_DIM: 512
    SCALE: 128
  LOSSES:
    CE:
      ALPHA: 0.3
      EPSILON: 0.1
      SCALE: 1.0
    FL:
      ALPHA: 0.25
      GAMMA: 2
      SCALE: 1.0
    NAME: ('CrossEntropyLoss', 'TripletLoss')
    TRI:
      HARD_MINING: True
      MARGIN: 0.3
      NORM_FEAT: False
      SCALE: 1.0
      USE_COSINE_DIST: False
  META_ARCHITECTURE: Baseline
  OPEN_LAYERS: 
  PIXEL_MEAN: [123.675, 116.28, 103.53]
  PIXEL_STD: [58.395, 57.120000000000005, 57.375]
  WEIGHTS: 
OUTPUT_DIR: logs/market1501/bagtricks
SOLVER:
  BASE_LR: 0.00035
  BIAS_LR_FACTOR: 2.0
  CHECKPOINT_PERIOD: 2000
  DELAY_ITERS: 100
  ETA_MIN_LR: 3e-07
  FREEZE_ITERS: 0
  GAMMA: 0.1
  HEADS_LR_FACTOR: 1.0
  IMS_PER_BATCH: 64
  LOG_PERIOD: 200
  MAX_ITER: 28000
  MOMENTUM: 0.9
  OPT: Adam
  SCHED: WarmupMultiStepLR
  STEPS: (8000, 14000)
  WARMUP_FACTOR: 0.01
  WARMUP_ITERS: 2000
  WARMUP_METHOD: linear
  WEIGHT_DECAY: 0.0005
  WEIGHT_DECAY_BIAS: 0.0
TEST:
  EVAL_PERIOD: 2000
  IMS_PER_BATCH: 512
  PRECISE_BN:
    DATASET: Market1501
    ENABLED: False
    NUM_ITER: 300

**Comment by L1aoXingyu** [comment]
Hi,
thanks for your attention to this project. I have updated some code and config files, you can try it again.

The most important reason is that feature after bnneck is used to compute triplet loss, I have added an option to decide which feature is chosen.
https://github.com/JDAI-CV/fast-reid/blob/46228ce946283d68d2b8549df8c958131d7702d9/fastreid/modeling/heads/bnneck_head.py#L48

**Comment by L1aoXingyu** [comment]
If you don't have other problems, I will close this issue. @Theohhhu

**Comment by Theohhhu** [comment]
Here I notice the feature after bnneck is only adopted in test stage of AWG while in BoT and SBS cases it is not . Is there any reason behind it?

**Comment by L1aoXingyu** [comment]
Feature after bnneck will be adopted for id loss in all cases, but in BoT and AGW, feature before bnneck will be used for triplet loss. Feature after bnneck will be used for triplet loss in SBS, because we can get a better result in this way.

**Comment by L1aoXingyu** [comment]
If u have no other problems, I will close it @Theohhhu
